samples/cellular/nrf_cloud_mqtt_multi_service/src/application.c

```diff
@@ -343,20 +303,40 @@ void main_application_thread_fn(void)
 			K_SECONDS(CONFIG_SENSOR_SAMPLE_INTERVAL_SECONDS), K_FOREVER);
 
 		if (IS_ENABLED(CONFIG_TEMP_TRACKING)) {
-			double temp = -1;
-
-			if (get_temperature(&temp) == 0) {
-				LOG_INF("Temperature is %d degrees C", (int)temp);
-				(void)send_sensor_sample(NRF_CLOUD_JSON_APPID_VAL_TEMP, temp);
-
-				monitor_temperature(temp);
+			struct nrf_cloud_sensor_data data = {
+				.app_id = NULL,
+				.type = NRF_CLOUD_SENSOR_TEMP,
+				.data_type = NRF_CLOUD_DATA_TYPE_DOUBLE,
+			};
+
+			if (get_temperature(&data.double_val) == 0) {
+				LOG_INF("Temperature is %d degrees C", (int)data.double_val);
+				LOG_DBG("Sending temperature...");
+				(void)send_device_message(&data);
+				LOG_DBG("Monitor temperature...");
+				monitor_temperature(data.double_val);
 			}
 		}
 
 		if (IS_ENABLED(CONFIG_TEST_COUNTER)) {
-			LOG_INF("Sent test counter = %d", counter);
-			(void)send_sensor_sample("COUNT", counter++);
+			struct nrf_cloud_sensor_data data = {
+				.app_id = "COUNT",
+				.data_type = NRF_CLOUD_DATA_TYPE_INT,
+				.int_val = counter++
+			};
+
+			LOG_INF("Sent test counter = %d", data.int_val);
+			(void)send_device_message(&data);
+		}
+
+#if defined(CONFIG_NRF_CLOUD_COAP)
+#if defined(CONFIG_NRF_CLOUD_COAP_FOTA)
+		if (handle_fota_process() != -EAGAIN) {
```

Comment:
TODO: allow this and shadow checking to be called after multiple sensor updates (lower frequency).

Alternatively, rework this loop to use a timer or deferred worker to sample everything at a separately configurable rate.

Reply:
FOTA is not connection management.

The shadow is multiple things, including connection management, but also declaration of supported sensors, which are managed here.

My only concern with using threads everywhere is that each one needs a stack, and that consumes RAM.  But yes, I know that was your design.

Reply:
done for both